src/System.Memory/tests/Memory/CustomMemoryForTest.cs

```diff
@@ -51,11 +51,24 @@ public override MemoryHandle Pin(int byteOffset = 0)
         {
             unsafe
             {
-                Retain();
-                if (byteOffset != 0 && (((uint)byteOffset) - 1) / Unsafe.SizeOf<T>() >= _array.Length)
-                    throw new ArgumentOutOfRangeException(nameof(byteOffset));
-                var handle = GCHandle.Alloc(_array, GCHandleType.Pinned);
-                return new MemoryHandle(this, Unsafe.Add<byte>((void*)handle.AddrOfPinnedObject(), _offset + byteOffset), handle);
+                Retain(); // this checks IsDisposed
+
+                try
+                {
+                    if ((IntPtr.Size == 4 && (uint)byteOffset > (uint)_array.Length * (uint)Unsafe.SizeOf<T>())
```

Comment:
Can you also make this change in corefxlab please?
https://github.com/dotnet/corefxlab/blob/d38d2ae7b5bb10036ff9c4ec1f400d5cb508fbd8/src/System.Buffers.Experimental/System/Buffers/Native/OwnedPinnedBuffer.cs#L62
https://github.com/dotnet/corefxlab/blob/1183c2a37f5c868487fc8a56c881fbe6482c339a/tests/System.Numerics.Tensors.Tests/NativeMemory.cs#L47
https://github.com/dotnet/corefxlab/blob/fc0c8d81d064d5af7b4ef65678d79dea16fbed4f/tests/System.Buffers.Primitives.Tests/CustomMemoryForTest.cs#L43
https://github.com/dotnet/corefxlab/blob/fd72e1afcc8ef9e6c365f1959933316b0ee2d2b9/src/System.Buffers.Experimental/System/Buffers/OwnedArray.cs#L39
https://github.com/dotnet/corefxlab/blob/fc0c8d81d064d5af7b4ef65678d79dea16fbed4f/tests/System.Buffers.Experimental.Tests/MemoryTests.cs#L222

Reply:
Sure. We're considering a tweak to the `Pin` API anyway, so that can probably all go in together as we're reacting to the changes.